include/boost/math/distributions/cauchy.hpp

```diff
@@ -275,29 +276,30 @@ inline RealType quantile(const complemented2_type<cauchy_distribution<RealType,
 
 template <class RealType, class Policy>
 inline RealType mean(const cauchy_distribution<RealType, Policy>&)
-{  // There is no mean:
```

Comment:
Apologies for being absent while this was being developed, just trying to catch up now....

We really shouldn't remove the static_asserts here: the point is that the cauchy distribution has no mean, so calling `mean` on such a distribution is almost certainly a programming error that should be caught early (rather than at runtime).  There are a small number of situations where we're writing generic software which "works with any distribution" where we can change the Policy as a "get out of jail free card" to convert the compile time error to a runtime one.

Reply:
I'm really glad to hear from you John. I was extremelty unsure about this. I came to the conclusion that (via default template parameters) the exact `policy` in question that was being used did not actually have any of the attributes needed for the `static_assert`.

These `static_assert`s are being asserted on [`defualt_policy`](https://github.com/boostorg/math/blob/d4a42f930e92cc2ad4ca0d99d323f1159493c4ae/include/boost/math/policies/policy.hpp#L223), which is empty.

So that thing either needs content, or the static assertions are bogus.

I'm not entirely sure of this, but seem to have gleaned this from the code, ...

Reply:
To be clear, the default is that these functions *do* static_assert, we should have added a test to that effect.... somewhere lost in the depths of time, I remember Paul and I having a long discussion about this, the "obvious" thing to do was to simply not provide functions for mean etc when those are undefined, but that then prevents the option of writing generic code.  So these need either a policy that contains `assert_undefined<false>`, or we need a `#define BOOST_MATH_ASSERT_UNDEFINED_POLICY false`.

Reply:
Ahh, instantate.hpp includes the define at it's start, but that will not be triggered if policy.hpp has been included first... so I'm guilty of some fragile programming there :(

Reply:
We could make the default policy have a bunch of constexpr stuff to pass static asserts. Or we could handle on the top layer with compiler switches.

I expect we will find several things of this nature as we press on toward ninety five, more percent.

This code base is awesome. Rarely, have I covereed stuff with such depth and detail and established nature and found such deep thought going into it.

But we will find a few thingy-ies needing tweaks and patches toward, 93, 95, 98, 99

Reply:
> Ahh, `instantate.hpp` includes the define at it's start, but that will not be triggered if `policy.hpp` has been included first, ...

Yeah! That is what I was missing. I studied that code for a while but did not dial that nuance.

Can we repair this in this running PR? Or should i take another run at it. I really got a good start at this thing...

Reply:
In other words @jzmaddock I'm looking to get about 1,500 lines covered if we can get these istantiations compiling and running in all situations. We are close. Maybe I need to include that policy and then restore those `static_asserts`. We are close to that in my latest PR.